rootston/output.c

```diff
@@ -512,6 +532,31 @@ static void render_output(struct roots_output *output) {
 
 damage_finish:
 	pixman_region32_fini(&damage);
+
+	// Send frame done events to all surfaces
+	if (output->fullscreen_view != NULL) {
+		if (wlr_output->fullscreen_surface == output->fullscreen_view->wlr_surface) {
+			// The surface is managed by the wlr_output
+			return;
+		}
+
+		view_for_each_surface(view, surface_send_frame_done, &data);
+
+#ifdef WLR_HAS_XWAYLAND
+		if (view->type == ROOTS_XWAYLAND_VIEW) {
+			xwayland_children_for_each_surface(view->xwayland_surface,
+				surface_send_frame_done, &data);
+		}
+#endif
+	} else {
+		struct roots_view *view;
```

Comment:
(nitpick) there already is a struct roots_view *view 40 lines earlier, although that's in the middle of the code, so not quite sure how fair it is to reuse it.
Conversely, that's what let the earlier re-use of view work; might make sense to delimit such 'local' variables without flow in an artificial block?